pkg/compose/publish.go

```diff
@@ -116,6 +151,94 @@ func (s *composeService) publish(ctx context.Context, project *types.Project, re
 	return nil
 }
 
+func processExtends(ctx context.Context, project *types.Project, extFiles map[string]string) ([]ocipush.Pushable, error) {
+	var layers []ocipush.Pushable
+	moreExtFiles := map[string]string{}
+	for xf, hash := range extFiles {
+		data, err := processFile(ctx, xf, project, moreExtFiles)
+		if err != nil {
+			return nil, err
+		}
+
+		layerDescriptor := ocipush.DescriptorForComposeFile(hash, data)
+		layerDescriptor.Annotations["com.docker.compose.extends"] = "true"
+		layers = append(layers, ocipush.Pushable{
+			Descriptor: layerDescriptor,
+			Data:       data,
+		})
+	}
+	for f, hash := range moreExtFiles {
+		if _, ok := extFiles[f]; ok {
+			delete(moreExtFiles, f)
+		}
+		extFiles[f] = hash
+	}
+	if len(moreExtFiles) > 0 {
+		extLayers, err := processExtends(ctx, project, moreExtFiles)
+		if err != nil {
+			return nil, err
+		}
+		layers = append(layers, extLayers...)
+	}
+	return layers, nil
+}
+
+func processFile(ctx context.Context, file string, project *types.Project, extFiles map[string]string) ([]byte, error) {
+	f, err := os.ReadFile(file)
+	if err != nil {
+		return nil, err
+	}
+
+	base, err := loader.LoadWithContext(ctx, types.ConfigDetails{
+		WorkingDir:  project.WorkingDir,
+		Environment: project.Environment,
+		ConfigFiles: []types.ConfigFile{
+			{
+				Filename: file,
+				Content:  f,
+			},
+		},
+	}, func(options *loader.Options) {
+		options.SkipValidation = true
+		options.SkipExtends = true
+		options.SkipConsistencyCheck = true
+		options.ResolvePaths = true
+	})
+	if err != nil {
+		return nil, err
+	}
+	for name, service := range base.Services {
+		if service.Extends == nil {
+			continue
+		}
+		xf := service.Extends.File
+		if xf == "" {
+			continue
+		}
+		if _, err = os.Stat(service.Extends.File); os.IsNotExist(err) {
+			// No local file, while we loaded the project successfully: This is actually a remote resource
+			continue
+		}
+
+		hash := fmt.Sprintf("%x.yaml", sha256.Sum256([]byte(xf)))
+		extFiles[xf] = hash
+
+		// FIXME implement without relying on yqlib so we don't get extra dependencies
+		exp := fmt.Sprintf(".services.%s.extends.file = %q", name, hash)
+		encoder := yqlib.NewYamlEncoder(yqlib.YamlPreferences{
+			ColorsEnabled: false,
+		})
+		decoder := yqlib.NewYamlDecoder(yqlib.YamlPreferences{})
+		out, err := yqlib.NewStringEvaluator().Evaluate(exp, string(f), encoder, decoder)
```

Comment:
TODO: I'm investigating if we could avoid ylib as a dependency and "just" reproduce this single search-and-replace feature

Reply:
Do you want to make the changes in this PR or in a follow up one?

Reply:
I'll look into this as a follow-up